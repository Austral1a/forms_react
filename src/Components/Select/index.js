export * from './Select.scss'
export * from './Select'
export * from './Option'
