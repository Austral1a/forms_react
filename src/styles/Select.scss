.select {
  width: 100%;
  padding: 4px 7px;
  border-radius: 4px;
}
