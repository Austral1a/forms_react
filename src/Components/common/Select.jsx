import React from 'react'
import '../../styles/Select.scss'

function Select({defaultValue, children, onChange}) {
    return (
        <select onChange={onChange} className='select' defaultValue={defaultValue}>
            {children}
        </select>
    )
}

export default Select
