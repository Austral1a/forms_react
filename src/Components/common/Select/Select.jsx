import React from 'react'
import classNames from 'classnames/dedupe'
import './Select.scss'

function Select({defaultValue, children, onChange, className}) {
    const customClasses = classNames('select', className)
    return (
        <select onChange={onChange} className={customClasses} defaultValue={defaultValue ? defaultValue : null}>
            {children}
        </select>
    )
}

export default Select
