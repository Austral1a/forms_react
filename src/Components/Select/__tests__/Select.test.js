import {Select} from '../Select';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';

describe("Select component", () => {
    it('Select must match snapshot', () => {
        const wrapper = shallow(<Select />)
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it("must render children", () => {
        const children = <option value="1">1</option>;
        const wrapper = shallow(<Select>{children}</Select>);

        expect(wrapper.find('select').contains(children)).toBeTruthy();
    });

    describe("defaultValue prop", () => {
        it("must set 'defaultValue' prop", () => {
            const defValue = 'us';
            const wrapper = shallow(<Select defaultValue={defValue} />);

            expect(wrapper.find('select').prop('defaultValue')).toBe(defValue);
        });

        it("must not set 'defaultValue' prop", () => {
            const wrapper = shallow(<Select />);

            expect(wrapper.find('select').prop('defaultValue')).toBe(null);
        });
    });

    it("must accurate manage 'onChange' prop", () => {
        const newValue = 'ua';
        const initValue = 'us';
        const state = {
            value: initValue
        };

        const onChange = () => {
            state['value'] = newValue
        };

        const wrapper = shallow(<Select onChange={onChange} />)
        expect(state.value).toBe(initValue);
        wrapper.find('select').simulate('change');
        expect(state.value).toBe(newValue);
    });
});
