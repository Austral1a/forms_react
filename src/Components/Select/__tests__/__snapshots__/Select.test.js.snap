// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select component Select must match snapshot 1`] = `
<select
  className="select"
  defaultValue={null}
/>
`;
